addons/mail/static/src/components/discuss/discuss_tests.js

```diff
@@ -421,6 +421,181 @@ QUnit.test('sidebar: add channel', async function (assert) {
     );
 });
 
+QUnit.test('sidebar: unpin channel', async function (assert) {
+    assert.expect(15);
+
+    Object.assign(this.data.initMessaging, {
+        channel_slots: {
+            channel_channel: [{
+                channel_type: "channel",
+                id: 20,
+                name: "General",
+            }],
+        },
+    });
+
+    await this.start({
+        async mockRPC(route, args) {
+            if (route === '/web/dataset/call_kw/mail.channel/execute_command') {
+                assert.deepEqual(args.args[0], [20], "The right id is sent to the server to remove");
+                assert.strictEqual(args.args[1], 'leave', "The right command is sent to the server");
+                return Promise.resolve();
+            }
+            return this._super(...arguments);
+        },
+    });
+
+    assert.containsOnce(
+        document.body,
+        '.o_Discuss_content .o_Discuss_thread[data-thread-local-id="mail.thread_mail.box_inbox"]',
```

Comment:
I personally dislike the use of data-, I think tests should test what matters functionally (eg. displaying the proper channel name as text).